### PR TITLE
Add support to net_map plugin for configuring Multus routes

### PR DESCRIPTION
**Pull request by lewisdenny**
Jira: https://issues.redhat.com/browse/OSPRH-7259

As a pull request owner and reviewers, I checked that:
- [x] Appropriate testing is done and actually running
- [x] Appropriate documentation exists and/or is up-to-date:
  - [x] Content of the docs/source is reflecting the change

Todo:
- [x] Add support for IPV6
- [x] Test in VA deployment with Octavia enabled

### CODE REVIEW

**Comment by lewisdenny** [comment]
/hold
Currently being tested D/S

File: tests/unit/module_utils/net_map/test_networking_definitions_network.py [hunk 1]
```diff
@@ -794,3 +794,41 @@ def test_network_definition_parse_valid_router_ok():
...
+                    "ranges": [
+                        {
+                            "start": str(network_ip[100]),
+                            "length": 20,
+                        },
+                    ],
+                    "routes": [
+                        {
+                            "destination": "192.168.121.0/24",
+                            "gateway": "192.168.122.1/32",
```
Comment (beagles) [inline tests/unit/module_utils/net_map/test_networking_definitions_network.py:821]: The gateway should be just the IP address and not include the mask. This causes an ipam parser error.

**Review by beagles — CHANGES_REQUESTED** [review]
The gateway is just an IP and doesn't need a mask so in th

File: tests/unit/module_utils/net_map/test_networking_definitions_network.py (hunk 1 above)
Comment (lewisdenny) [reply tests/unit/module_utils/net_map/test_networking_definitions_network.py:821]: Ah mate, I checked this one when I was implementing. It seemed like it wanted the mask: https://github.com/containernetworking/plugins/blob/acf8ddc8e1128e6f68a34f7fe91122afeb1fa93d/plugins/ipam/static/static_test.go#L59

I will check with you on the error, sorry about that one!

Comment (lewisdenny) [reply tests/unit/module_utils/net_map/test_networking_definitions_network.py:821]: Oh `s/destination/gateway/` got it!

Comment (lewisdenny) [reply tests/unit/module_utils/net_map/test_networking_definitions_network.py]: This should be resolved now in the latest revision so resolving

**Review by beagles — APPROVED** [review]

**Comment by beagles** [comment]
> This is now passing downstream in a UNI* job with Octavia enabled. This patch is functional and ready to merge from my side, @beagles could you please confirm when ready if your okay?

Tested and seems good to go!

> 
> @pablintino from my team will review the code :)

File: plugins/module_utils/net_map/networking_definition.py [hunk 2]
```diff
@@ -471,6 +471,26 @@ def check_host_network_ranges_collisions(
     return None, None
 
 
+def _get_field_ip_net_version(raw_field: str) -> typing.Union[int, None]:
```
Comment (pablintino) [inline plugins/module_utils/net_map/networking_definition.py:474]: I need to update other usages of Union, but I started using another cleaner approach, `typing.Optional[int]`.

Comment (lewisdenny) [reply plugins/module_utils/net_map/networking_definition.py:474]: Ooh `Optional[X] is equivalent to X | None (or Union[X, None]).` very nice man.  [1]
[1] https://docs.python.org/3/library/typing.html#typing.Optional

**Pull request merged**